spectrochempy/analysis/_base.py

```diff
@@ -1043,6 +1085,446 @@ def Y(self):
         return Y
 
 
+# ======================================================================================
+# Base class CrossDecompositionAnalysis
+# ======================================================================================
+
+
+class CrossDecompositionAnalysis(DecompositionAnalysis):
+    """
+    Abstract class to write analysis cross decomposition models such as `PLS`, ...
+
+    Subclass this to get a minimal structure
+
+    See Also
+    --------
+    PLSRegression : Perform a Partial Least Square Regression .
+    """
+
+    # This class is a subclass of DecompositionAnalysis, so we define only additional
+    # attributes and methods necessary for cross decomposition model.
+
+    # Get doc sections for reuse in subclass
+    _docstring.get_sections(
+        _docstring.dedent(__doc__),
+        base="CrossDecompositionAnalysis",
+        sections=["See Also"],
+    )
+
+    # ----------------------------------------------------------------------------------
+    # Public methods
+    # ----------------------------------------------------------------------------------
+
+    @_wrap_ndarray_output_to_nddataset(meta_from="_Y", title=None)
+    @_docstring.dedent
+    def predict(self, X=None, copy=True):
+        """
+        Predict targets of given observations.
+
+        Parameters
+        ----------
+        X : `NDDataset` or :term:`array-like` of shape (:term:`n_observations`\ , :term:`n_features`\ ), optional
+            New data, where :term:`n_observations` is the number of observations
+            and :term:`n_features` is the number of features.
+            if not provided, the input dataset of the `fit` method will be used.
+        copy : `bool`\ , default: `True`
+            Whether to copy X and Y, or perform in-place normalization.
+
+        Returns
+        -------
+        `NDDataset`
+            Datasets with shape (:term:`n_observations`\ ,) or ( :term:`n_observations`\ , :term:`n_targets` \ ).
+        """
+        if not self._fitted:
+            raise NotFittedError()
+
+        if X is None:
+            X = self._X_preprocessed
+        elif isinstance(X, NDDataset):
+            X = X.data
+
+        return self._predict(X, copy)
```

Comment:
I would suppress the copy parameter and replace here copy by self.copy.

Reply:
duly noted (see reply above)

Reply:
Same thing for the transform method. 
In addition copy is not defined for fit_transform, which is not consistent!